app/controllers/api/account.php

```diff
@@ -1677,6 +1677,96 @@
+    ->param('providerType', '', new WhiteList(['email', 'sms', 'push']), 'The target provider type. Can be one of the following: `email`, `sms` or `push`.')
+    ->param('identifier', '', new Text(Database::LENGTH_KEY), 'The target identifier (token, email, phone etc.)')
+    ->param('providerId', '', new UID(), 'Provider ID. Message will be sent to this target from the specified provider ID. If no provider ID is set the first setup provider will be used.', true)
+    ->param('name', '', new Text(256), 'Set Client device name manually instead of using user agent. Used to identify the client device make, model, OS, etc.', true)
+    ->inject('queueForEvents')
+    ->inject('user')
+    ->inject('request')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (string $targetId, string $providerType, string $identifier, string $providerId, string $name, Event $queueForEvents, Document $user, Request $request, Response $response, Database $dbForProject) {
+        $targetId = $targetId == 'unique()' ? ID::unique() : $targetId;
+
+        $provider = new Document();
+
+        if ($providerType === 'push') {
+            $provider = Authorization::skip(fn () => $dbForProject->getDocument('providers', $providerId));
+
+            if ($provider->isEmpty()) {
+                throw new Exception(Exception::PROVIDER_NOT_FOUND);
+            }
+        }
+
+        if ($user->isEmpty()) {
+            throw new Exception(Exception::USER_NOT_FOUND);
+        }
+
+        $target = Authorization::skip(fn () => $dbForProject->getDocument('targets', $targetId));
+
+        if (!$target->isEmpty()) {
+            throw new Exception(Exception::USER_TARGET_ALREADY_EXISTS);
+        }
+
+        $detector = new Detector($request->getUserAgent());
+        $detector->skipBotDetection(); // OPTIONAL: If called, bot detection will completely be skipped (bots will be detected as regular devices then)
+
+        $os = $detector->getOS();
+        $client = $detector->getClient();
+        $device = $detector->getDevice();
+
+        try {
+            $target = $dbForProject->createDocument('targets', new Document([
+                '$id' => $targetId,
+                '$permissions' => [
+                    Permission::read(Role::user($user->getId())),
+                    Permission::update(Role::user($user->getId())),
+                ],
+                'providerId' => $providerId ?? null,
+                'providerInternalId' => $provider->getInternalId() ?? null,
+                'providerType' =>  $providerType,
+                'userId' => $user->getId(),
+                'userInternalId' => $user->getInternalId(),
+                'identifier' => $identifier,
+                'name' => $name ?? [
```

Comment:
Can you test with different devices (or just user-agents, I think that's how the detector works) and see what "{$device['deviceBrand']} {$device['deviceName']} {$device['deviceModel']}" looks like? Ideally we get  `Samsung Galaxy S23` or `Apple iPhone 14`

Reply:
``` 

array(3) {
  ["osCode"]=>
  string(3) "IOS"
  ["osName"]=>
  string(3) "iOS"
  ["osVersion"]=>
  string(4) "16.6"
}
array(6) {
  ["clientType"]=>
  string(7) "browser"
  ["clientCode"]=>
  string(2) "MF"
  ["clientName"]=>
  string(13) "Mobile Safari"
  ["clientVersion"]=>
  string(4) "16.6"
  ["clientEngine"]=>
  string(6) "WebKit"
  ["clientEngineVersion"]=>
  string(8) "605.1.15"
}
array(3) {
  ["deviceName"]=>
  string(10) "smartphone"
  ["deviceBrand"]=>
  string(5) "Apple"
  ["deviceModel"]=>
  string(6) "iPhone"
}

```

Reply:
removed name and kept only brand and model